### PR TITLE
Created operations from partition-aware-factory when OperationFactory…

### PR DESCRIPTION
**Pull request by ahmetmircik**
…Wrapper is a wrapper for it

fixes https://github.com/hazelcast/hazelcast-enterprise/issues/1213

### CODE REVIEW

**Comment by ahmetmircik** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/InvokeOnPartitions.java [hunk 1]
```diff
@@ -124,7 +126,7 @@ private void retryFailedPartitions() throws InterruptedException, ExecutionExcep
 
         for (Integer failedPartition : failedPartitions) {
             Operation operation;
-            if (operationFactory instanceof PartitionAwareOperationFactory) {
+            if (isPartitionAwareFactory(operationFactory)) {
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/InvokeOnPartitions.java]: Why not add a method 'isPartitionAware' to the OperationFactory interface?

Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/InvokeOnPartitions.java]: Yes good point, here my goal is to fix this issue: hazelcast/hazelcast-enterprise#1213 on what we have already and porting it to maintenance. `PartitionAwareOperationFactory` and `OperationFactory` relation can be handled in a separate issue. Does it make sense?

Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/InvokeOnPartitions.java]: created an issue for this https://github.com/hazelcast/hazelcast/issues/9555

**Comment by pveentjer** [comment]
:+1:  But we should make a note somewhere the design should be cleaned up.

**Pull request merged**